src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -74,6 +97,199 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
+                    email: orderDetails.payer.email_address,
+                    address1: orderDetails.purchase_units[0].shipping.address.address_line_1,
+                };
+                const lineItems = this._collectLineItems(cart.lineItems);
+                const consignment = [{
+                    shippingAddress,
+                    lineItems,
+                }];
+                await this._store.dispatch(this._consignmentActionCreator.createConsignments(consignment));
+                const updatedBillingAddress = await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+                const consignments = updatedBillingAddress.consignments.getConsignmentsOrThrow();
+
+                if (consignments && this._selectedShippingOptionId) {
+                    await this._store.dispatch(this._consignmentActionCreator.updateConsignment({id: consignments[0].id, shippingOptionId: this._selectedShippingOptionId}));
+                }
+                if (this._orderActionCreator) {
+                    const submitOrderPayload = {};
+                    const submitOrderOptions = {
+                        params: {
+                            methodId: this._paymentMethod?.id,
+                        },
+                    };
+                    await this._store.dispatch(this._orderActionCreator.submitOrder(submitOrderPayload, submitOrderOptions));
+
+                    const paymentData =  {
+                        formattedPayload: {
+                            vault_payment_instrument: null,
+                            set_as_default_stored_instrument: null,
+                            device_info: null,
+                            method_id: this._paymentMethod?.id,
+                            paypal_account: {
+                                order_id: data.orderID,
+                            },
+                        },
+                    };
+                    await this._store.dispatch(this._paymentActionCreator.submitPayment({ ...{methodId: 'paypalcommerce'}, paymentData }));
+                }
+                window.location.assign('/checkout/order-confirmation');
+            } catch (e) {
+                throw new RequestError(e);
+            }
+        }
+
+    }
+
+    private async _onShippingChangeHandler(data: ShippingChangeData, actions: ApproveActions, cart: Cart) {
+        const baseOrderAmount = cart.baseAmount;
+        let shippingAmount = '0.00';
+        this._currentShippingAddress = await this._transformToAddress(data.shipping_address);
+        const lineItems = this._collectLineItems(cart.lineItems);
+        const consignment = [{
+            shippingAddress: {...this._currentShippingAddress},
+            lineItems,
+        }];
+        await this._store.dispatch(this._consignmentActionCreator.createConsignments(consignment));
+        const checkout = this._store.getState().checkout.getCheckoutOrThrow();
```

Comment:
Per our discussion with @koa75, the patch should be performed on the backend side in the scope of initialization entitiy on the bigpay side (it will be working with the quote).